tests/mock_crust/utils.rs

```diff
@@ -295,20 +296,26 @@ pub fn poll_all(nodes: &mut [TestNode], clients: &mut [TestClient]) -> bool {
 /// Polls and processes all events, until there are no unacknowledged messages left and clearing
 /// the nodes' state triggers no new events anymore.
 pub fn poll_and_resend(nodes: &mut [TestNode], clients: &mut [TestClient]) {
+    // Expect advance at least three times, given NODE_CONNECT_TIMEOUT_SECS is 60s and
+    // ACK_TIMEOUT_SECS is 20s
+    let mut num_of_advance = 0;
     for _ in 0..MAX_POLL_CALLS {
         if poll_all(nodes, clients) {
             let mut call_count = 1;
-            while resend_unacknowledged(nodes, clients) && poll_all(nodes, clients) {
+            while poll_all(nodes, clients) {
```

Comment:
Does it still make sense to have two nested loops that both call `poll_all` now?